File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:

        hm = {}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in hm:
                return [hm[diff],i]
            hm[n]=i

        return
        

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    std::vector<int> twoSum(const std::vector<int>& nums, int target) {
        std::unordered_map<int, int> numMap;
        
        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];
            
           
            if (numMap.find(complement) != numMap.end()) {
                return {numMap[complement], i}; 
            }
            
            // Add current number to the map
            numMap[nums[i]] = i;
        }
        
        return {}; 
    }
};